### PR TITLE
test(hits): first pass at widgets testing

### PR DESCRIPTION
**Pull request by vvo**
@redox, here's how I did it.

I removed the pre-bound Template component because it in fact
complexifies a lot testing and is not more easy to understand than
what I did here I believe.

What do you think?

(And no, it seems react contexts will not be more easy to use than
what we have here).

### CODE REVIEW

**Comment by vvo** [comment]
cc @algolia/javascript what do you think? I propose a new way to pass templateProps down the line.

The previous version is fun but really hard to test because it requires more and more mocking while not providing a code that is easier to understand.

File: widgets/hits.js [hunk 1]
```diff
@@ -3,9 +3,6 @@ var ReactDOM = require('react-dom');
 
 var utils = require('../lib/utils.js');
 
-var bindProps = require('../decorators/bindProps');
```
Comment (vvo) [inline widgets/hits.js]: less dependencies = less coupling, easier to test

Comment (redox) [inline widgets/hits.js]: :+1: and much more clear :100:

**Comment by redox** [comment]
LGTM :+1:

File: widgets/hits.js (hunk 1 above)
Comment (vvo) [inline widgets/hits.js]: > and much more clear

Ok so I will do this for all other widgets while I am testing them one after one

Comment (redox) [inline widgets/hits.js]: > Ok so I will do this for all other widgets while I am testing them one after one

Cool, thx @vvo

**Comment by vvo** [comment]
Ok I have updated the PR @redox with even more separation. Have a look for your own tests.

**Comment by redox** [comment]
:+1: Go!

**Comment by pixelastic** [comment]
Oh thanks. The prebound template was really hard to grasp and ugly

**Pull request merged**